include/dos_inc.h

```diff
@@ -645,25 +629,32 @@ class DOS_FCB : public MemStruct {
 	#endif
 };
 
-class DOS_MCB : public MemStruct{
+/* Memory Control Block */
+
+class DOS_MCB : public MemStruct {
 public:
-	DOS_MCB(Bit16u seg) { SetPt(seg); }
+	DOS_MCB(uint16_t seg) : MemStruct(seg, 0) {}
+
 	void SetFileName(char const * const _name) { MEM_BlockWrite(pt+offsetof(sMCB,filename),_name,8); }
 	void GetFileName(char * const _name) { MEM_BlockRead(pt+offsetof(sMCB,filename),_name,8);_name[8]=0;}
-	void SetType(Bit8u _type) { sSave(sMCB,type,_type);}
-	void SetSize(Bit16u _size) { sSave(sMCB,size,_size);}
-	void SetPSPSeg(Bit16u _pspseg) { sSave(sMCB,psp_segment,_pspseg);}
-	Bit8u GetType(void) { return (Bit8u)sGet(sMCB,type);}
-	Bit16u GetSize(void) { return (Bit16u)sGet(sMCB,size);}
-	Bit16u GetPSPSeg(void) { return (Bit16u)sGet(sMCB,psp_segment);}
+
+	void SetType(uint8_t type) { SSET_BYTE(sMCB, type, type); }
+	uint8_t GetType() const { return SGET_BYTE(sMCB, type); }
+
+	void SetSize(uint16_t size) { SSET_WORD(sMCB, size, size); }
+	uint16_t GetSize() const { return SGET_WORD(sMCB, size); }
```

Comment:
Suggest similarly rename the arguments to differentiate them from the members. (ie: "size, size" and "type, type").